mlflow/store/artifact/hdfs_artifact_repo.py

```diff
@@ -192,7 +192,8 @@ def hdfs_system(scheme, host, port):
 
     host = scheme + "://" + host if host else "default"
 
-    if packaging.version.parse(pyarrow.__version__) < packaging.version.parse("2.0.0"):
+    pyarrow_version = packaging.version.parse(pyarrow.__version__)
+    if pyarrow_version.major >= 2:
```

Comment:
```suggestion
    if pyarrow_version.major < 2:
```

@harupy Was the previous behavior reversed on accident? The previous logic was to only use `pyarrow.fs.HadoopFileSystem` if `pyarrow`'s version is < 2. The proposed change seems to reverse that logic. Is this intentioonal?

Reply:
@dbczumar yes, should be correct.

Reply:
note `pyarrow.fs.HadoopFileSystem` != `pyarrow.hdfs.HadoopFileSystem`.